### PR TITLE
docs: add k8s dashboard recipe

### PR DESCRIPTION
**Pull request by desimone**
## Summary
Adds a doc recipe showing how to protect Kubernetes dashboard with Pomerium. Includes installation guides for helm, Nginx ingress, dashboard, pomerium, and cert-manager. 

Main purpose is to compare and contrast access control via forward-authentication and direct proxying. 

Here's the [current live preview of the doc](https://deploy-preview-365--pomerium-docs.netlify.com/recipes/kubernetes.html)

## Related issues

- #94 
- #323 

**Checklist**:
- [x] updated docs
- [x] ready for review

### CODE REVIEW

**Comment by desimone** [comment]
\cc @victornoel @lukasmrtvy @pecigonzalo @tlvenn would welcome any feedback on this doc relative to how you are using `forward-auth`

**Comment by victornoel** [comment]
@desimone I read it quickly, but I don't get what are the two last screenshots meant to demonstrate, they seem to show the same thing exactly and to be not related to the accompanying text

**Comment by pecigonzalo** [comment]
Looks interesting, ill give it a go later this week

**Comment by lukasmrtvy** [comment]
@desimone @victornoel exactly, not sure about last two screenshots.. I am usually deploying https://github.com/kubernetes/dashboard/blob/master/aio/deploy/alternative.yaml ( no TLS, no login ), with linked service account with admin cluster role.

I have to check what exactly api server can log...  ( for audit purposes ),  headers from pomerium would be great..

**Comment by desimone** [comment]
Thanks for taking a look 👍  I think I'm trying to show too much here so it's not obvious:

1 ) how to set up kubernetes dashboard
2) how to use pomerium in a forward-auth configuration to handle access
3) how to use pomerium's proxy directly to handle access

2,3 function almost identically in practice, but in 3) you can inject authentication headers directly skipping the the login page.

File: docs/recipes/kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+```sh
+NAME                    HOSTS                                                ADDRESS   PORTS     AGE
+dashboard-fwdauth       dashboard-fwdauth.domain.example                               80, 443   42h
+dashboard-proxied       dashboard-proxied.domain.example                               80, 443   42h
+helm-pomerium           *.domain.example,authenticate.domain.example                   80, 443   42h
+```
+
+## Comparing deployments
+
+Though the net result will be similar between using forward-auth and direct proxying, there are a few difference.
```
Comment (travisgroth) [inline docs/recipes/kubernetes.md]: >difference

differences

Comment (desimone) [reply docs/recipes/kubernetes.md]: 👍

**Comment by desimone** [comment]
> Those last screen shots might be better illustrated with video to show the login experience.

Added two comparison videos. Hopefully makes it a bit more clear. 

> Minor typo. Looks good otherwise. Good demonstration of showing Pomerium as a trusted mediator.

👍 

> Also weird that there's no way to have dashboard use the existing service account with no prompt - TIL that.

Ditto!

**Pull request merged**